pkg/handler/handler.go

```diff
@@ -33,14 +68,64 @@ func New(b *storage.BucketHandle, cacheDir, listenAddress string) *http.Server {
 	}
 }
 
-func (pbh *prowBucketHandler) router(resp http.ResponseWriter, r *http.Request) {
+func (pbh *prowBucketHandler) welcomeHandler(resp http.ResponseWriter, r *http.Request) {
 	if strings.HasPrefix(r.URL.String(), "/logs") {
-		pbh.handleLogRequest(resp, r)
-		return
+		// save the URl for logs so that we can redirect to this later
+		pbh.requestURL = r.URL.String()
+
+		if isLoggedIn(r) {
+			pbh.handleLogRequest(resp, r)
+			return
+		}
+
+		// if the user is not logged in, ask them to login
```

Comment:
This is not quite right, as we have to check whether the logs are publicly available or not

Reply:
Althought maybe it makes sense to require everyone to be logged in and do an additional authz checks for private logs in order to prevent random scape bots to download our logs?

Reply:
Updated in https://github.com/kubermatic/tail/pull/4.